front/src/routes/scene/edit-scene/actions/LightFadeInParams.jsx

```diff
@@ -0,0 +1,137 @@
+    })
+  }
+  getOptions = async () => {
+    try {
+      const devices = await this.props.httpClient.get('/api/v1/device', {
+        device_feature_category: 'light',
+        device_feature_type: 'brightness'
+      });
+      const deviceOptions = devices.map(device => ({
+        value: device.selector,
+        label: device.name
+      }));
+      await this.setState({ deviceOptions });
+      this.refreshSelectedOptions(this.props);
+      return deviceOptions;
+    } catch (e) {
+      console.log(e);
+    }
+  };
+  handleChange = selectedOptions => {
+    if (selectedOptions) {
+      const lights = selectedOptions.map(selectedOption => selectedOption.value);
+      this.props.updateActionProperty(this.props.columnIndex, this.props.index, 'devices', lights);
+    } else {
+      this.props.updateActionProperty(this.props.columnIndex, this.props.index, 'devices', []);
+    }
+  };
+  refreshSelectedOptions = nextProps => {
+    const selectedOptions = [];
+    if (nextProps.action.devices && this.state.deviceOptions) {
+      nextProps.action.devices.forEach(light => {
+        const deviceOption = this.state.deviceOptions.find(deviceOption => deviceOption.value === light);
+        if (deviceOption) {
+          selectedOptions.push(deviceOption);
+        }
+      });
+    }
+    this.setState({ selectedOptions });
+  };
+  constructor(props) {
+    super(props);
+    this.state = {
+      deviceOptions: null,
+      selectedOptions: [],
+      parameters: props.parameters
+    };
+  }
+  async componentDidMount() {
+    this.getOptions();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.refreshSelectedOptions(nextProps);
+  }
+
+  render(props, { selectedOptions, deviceOptions }) {
+    const { parameters } = props.action;
+    return (
+      <div class="form-group">
+        <label class="form-label">
+          <Text id="editScene.actionsCard.fadeInLights.label" />
+        </label>
+        <Select
+          defaultValue={[]}
+          isMulti
+          value={selectedOptions}
+          onChange={this.handleChange}
+          options={deviceOptions}
+        />
+        <label className="form-label">
+          <Text id="editScene.actionsCard.fadeInLights.targetBrightness" />
+        </label>
+        <input
+          style={{
+            minHeight: '30px'
+          }}
+          type="range"
+          value={parameters ? parameters.targetBrightness : undefined}
+          onChange={this.handleChangeTargetBrightness}
+          className="form-control custom-range"
+          step="1"
+          min="0"
+          max="100"
+        />
+        <div className="row">
+          <div className="col-md-6">
+            <Localizer>
+              <input
+                type="text"
+                className="form-control"
+                value={parameters ? parameters.durationValue : undefined}
```

Comment:
To let you know, you case use `get-value` (by imorting it `import 'get' from 'get-value'`) and do
`get(parameters, 'durationValue')`

or if you need default value `get(parameters, 'durationUnit', { default: 'sesonds'})`

https://www.npmjs.com/package/get-value

But works well as written now.

Reply:
I always prefer to use Vanilla JS instead of some module if the cost of it is acceptable. Would love to have elvis operator though.